matplotcheck/raster.py

```diff
@@ -100,68 +97,62 @@ def assert_legend_accuracy_classified_image(
-        which element of all_label_options matches that entry. E.g. if the
-        first legend entry has a match in the first list in all_label_options,
-        then that legend entry corresponds to the first class (value 0).
-        Then the plot image array is copied and the values are set to the
-        legend label that match the values (i.e. the element in
-        all_label_options). The same is done for the expected image array.
-        Finally those two arrays of strings are compared. Passes if they match.
         """
         # Retrieve image array
-        im_data = []
-        if self.ax.get_images():
-            im = self.ax.get_images()[0]
-            im_data, im_cmap = im.get_array(), im.get_cmap()
-        assert list(im_data), "No Image Displayed"
+        im_data = self.get_plot_image()
 
-        # Retrieve legend
-        legends = self.get_legends()
-        assert legends, "No legend displayed"
+        assert list(im_data), "No Image Displayed"
 
         # Retrieve legend entries and find which element of all_label_options
         # matches that entry
-        legend_dict = {}
-        for p in [
-            p
-            for sublist in [leg.get_patches() for leg in legends]
-            for p in sublist
-        ]:
-            label = p.get_label().lower()
-            legend_dict[p.get_facecolor()] = self._which_label(
-                label, all_label_options
-            )
 
-        # Check that each legend entry label is in one of all_label_options
-        assert len([val for val in legend_dict.values() if val]) == len(
-            all_label_options
-        ), "Incorrect legend labels"
-
-        # Create two copies of image array, one filled with the plot data class
-        # labels (im_data_labels) and the other with the expected labels
-        # (im_expected_labels)
-        im_class_dict = {}
-        for val in np.unique(im_data):
-            im_class_dict[val] = legend_dict[im_cmap(im.norm(val))]
-        im_data_labels = [
-            [im_class_dict[val] for val in row] for row in im_data.data
-        ]
-        im_expected_labels = [
-            [all_label_options[val][0] for val in row] for row in im_expected
+        labels = self.get_legend_labels()
+
+        assert len(labels) == len(all_label_options), (
+            "Number of label options provided doesn't match the number of"
+            " labels found in the image."
+        )
+
+        labels_check = [
+            self._which_label(label, all_label_options[i])
+            for i, label in enumerate(labels)
         ]
 
-        # Check that expected and actual labels match up
+        # Check that each legend entry label is in one of all_label_options
+        assert all(
+            labels_check
+        ), "Provided legend labels don't match labels found."
+
+        # Check that expected and actual arrays data match up
         assert np.array_equal(
```

Comment:
this test really doesn't get at what we need to do. it just tests that the image is correct which we do elsewhere. so it probably can be removed.